gobblin-kafka/src/main/java/gobblin/kafka/schemareg/KafkaSchemaRegistry.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.kafka.schemareg;
+
+import java.io.IOException;
+
+import org.apache.avro.Schema;
+
+
+/**
+ * An interface for a Kafka Schema Registry
+ * Classes implementing this interface will typically be constructed by a {@link KafkaSchemaRegistryFactory}
+ * and should have a constructor that takes a {@link java.util.Properties} object as a parameter.
+ *
+ * @param <K> : the type of the schema identifier (e.g. int, string, md5, ...)
+ *
+ */
+public interface  KafkaSchemaRegistry<K> {
```

Comment:
I plan to have this super simple interface replace gobblin.metrics.kafka.KafkaSchemaRegistry (which is a base class with a ton of behavior). 

LiKafkaSchemaRegistry will replace gobblin.metrics.kafka.KafkaSchemaRegistry and gobblin.metrics.kafka.KafkaAvroSchemaRegistry.